File: 1회차/programmers_algorithm/7.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(string word) {
    int answer = 0;
    char arr[5] = {'A', 'E', 'I', 'O', 'U'};
    vector<string> v;
    for (int i = 1; i <= 5; i++) {
        int comb = 1;
        for (int j = 0; j < i; j++) comb*=5;
        for (int j = 0; j < comb; j++) {
            int cmb = j; string str;
            for (int k = 0; k < i; k++) {
                str.push_back(arr[cmb%5]);
                cmb/=5;
            }
            v.push_back(str);
        }
    }
    sort(v.begin(),v.end());
    for (int i = 0; i < v.size(); i++) {
        if (word==v[i]) return i+1;
    }
}

File: 1회차/programmers_algorithm/7.py
def solution(word):
    arr=['A', 'E', 'I', 'O', 'U']
    v = []
    for i in range(1,6):
        comb=1
        for j in range(i):
            comb*=5
        for j in range(comb):
            cmb=j
            s = ''
            for k in range(i):
                s+=(arr[cmb%5])
                cmb=int(cmb/5)
            v.append(s)
    v.sort()
    for i in range(len(v)):
        if word==v[i]:
            return i+1